src/App.jsx

```diff
@@ -1,31 +1,31 @@
-import { Routes, Route } from "react-router-dom";
-import { PublicRoute } from "./routes";
-import { useEffect } from "react";
+import { Routes, Route } from 'react-router-dom';
+import { PublicRoute, ProtectedRoute } from './routes';
+import { useEffect } from 'react';
 import { GlobalStyles } from './styles/GlobalStyles.styled';
-import { FontStyles } from "./styles/FontStyles";
-import { useAuthStore } from "./store/auth";
+import { FontStyles } from './styles/FontStyles';
+import { useAuthStore } from './store/auth';
 
-import SharedLayout from "./components/SharedLayout";
-import RegisterPage from "./pages/RegisterPage";
-import LoginPage from "./pages/LoginPage";
-import ConditionsPage from "./pages/ConditionsPage";
-import CongratsPage from "./pages/CongratsPage";
-import PolicyPage from "./pages/PolicyPage";
-import NotFoundPage from "./pages/NotFoundPage";
-import ChangeEmailPage from "./pages/ChangeEmailPage";
-import ResetPasswordPage from "./pages/ResetPasswordPage";
-import ChangePasswordPage from "./pages/ChangePasswordPage";
-import CreateNewPasswordPage from "./pages/CreateNewPasswordPage";
-import CoursesPage from "./pages/CoursesPage";
+import SharedLayout from './components/SharedLayout';
+import RegisterPage from './pages/RegisterPage';
+import LoginPage from './pages/LoginPage';
+import ConditionsPage from './pages/ConditionsPage';
+import CongratsPage from './pages/CongratsPage';
+import PolicyPage from './pages/PolicyPage';
+import NotFoundPage from './pages/NotFoundPage';
+import ChangeEmailPage from './pages/ChangeEmailPage';
+import ResetPasswordPage from './pages/ResetPasswordPage';
+import ChangePasswordPage from './pages/ChangePasswordPage';
+import CreateNewPasswordPage from './pages/CreateNewPasswordPage';
+import CoursesPage from './pages/CoursesPage';
+import ProfilePage from './pages/ProfilePage';
+import MyAdvertPage from './pages/MyAdvertPage';
 
 
 const App = () => {
   const isAuth = useAuthStore((state) => state.isAuth);
```

Comment:
Please substitute this approach with this one:
```
const { isAuth, setToken, refresh, token } = useAuthStore();
```
And also, please follow this method across the whole project.

Reply:
has been fixed